src/Connection.cpp

```diff
@@ -64,6 +64,21 @@ Connection::Connection(Connection::BusType type, std::unique_ptr<ISdBus>&& inter
     loopExitFd_ = createProcessingLoopExitDescriptor();
 }
 
+Connection::Connection(const std::string& host, std::unique_ptr<ISdBus> &&interface)
+    : iface_(std::move(interface))
+    , busType_(BusType::eRemoteSystem)
+    , host_(host)
+{
+    assert(iface_ != nullptr);
+
+    auto bus = openBus(busType_);
+    bus_.reset(bus);
+
+    finishHandshake(bus);
+
+    loopExitFd_ = createProcessingLoopExitDescriptor();
```

Comment:
This code is identical to the other constructor and could be refactored and shared between both, if that's preferred. What's important is that we are able to set `host_` before `openBus()` is called (which makes forwarding this constructor to the other not possible)

Reply:
I will resolve the duplication in a different way, see general comment below.